db/db_impl_files.cc

```diff
@@ -313,6 +315,16 @@ void DBImpl::DeleteObsoleteFileImpl(int job_id, const std::string& fname,
     file_deletion_status =
         DeleteSSTFile(&immutable_db_options_, fname, path_id);
   } else {
+    if (type == kLogFile) {
+      // Before deleting the file, mark file as deleted in the manifest
+      VersionEdit edit;
+      edit.SetDeletedLogNumber(number);
+      auto edit_cfd = versions_->GetColumnFamilySet()->GetDefault();
+      auto edit_cf_opts = edit_cfd->GetLatestMutableCFOptions();
+      mutex_.Lock();
+      versions_->LogAndApply(edit_cfd, *edit_cf_opts, &edit, &mutex_);
+      mutex_.Unlock();
```

Comment:
We need to make the manifest file downgrade-able. The manifest file generated by a newer version needs to be able to be readable from an older version. We need to have an option to switch this on, which is by default off.

Another choice is to make use of customize tag in AddFile(), which is a little bit odd though, but we may be able to make it workable.

Reply:
Would it make sense for an old binary to ignore the unknown entries in the manifest generated by a newer binary?

Reply:
@maysamyabandeh we do have something like that as a field in NewFile entry: https://github.com/facebook/rocksdb/blob/v5.10.3/db/version_edit.cc#L131-L157. It is possible to build something like that in the future.

However, it doesn't solve the downgrade-able problem, unless we have the generic framework check in and have the new field check in after many releases. (It takes time to type all the details so we can discuss in person if you want)